src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,49 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ELEMENTS = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ELEMENTS];
+        values = (V[]) new Object[MAX_ELEMENTS];
+    }
+
     @Override
     public void put(K key, V value) {
+        boolean isNewKey = true;
+        for (int i = 0; i < size; i++) {
+            if (key == keys[i] || key != null && key.equals(keys[i])) {
+                values[i] = value;
+                isNewKey = false;
+                break;
+
+            }
+        }
+        if (isNewKey) {
+            keys[size] = key;
+            values[size] = value;
+            size++;
+        }
+
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == (key) || (keys[i] != null && !(keys[i].getClass().isPrimitive())
+                    && keys[i].equals(key))) {
+                return values[i];
+            }
+        }
```

Comment:
redundant verification
```suggestion
        for (int i = 0; i < size; i++) {
            if (keys[i] == key || (keys[i] != null && keys[i].equals(key))) {
                return values[i];
            }
        }
```